src/dc_array.rs

```diff
@@ -160,120 +160,81 @@ impl From<Vec<dc_location>> for dc_array_t {
 }
 
 pub unsafe fn dc_array_unref(array: *mut dc_array_t) {
-    if array.is_null() {
-        return;
-    }
+    assert!(!array.is_null());
     Box::from_raw(array);
 }
 
 pub unsafe fn dc_array_add_uint(array: *mut dc_array_t, item: uintptr_t) {
-    if !array.is_null() {
-        (*array).add_uint(item);
-    }
+    assert!(!array.is_null());
+    (*array).add_uint(item);
 }
 
 pub unsafe fn dc_array_add_id(array: *mut dc_array_t, item: uint32_t) {
-    if !array.is_null() {
-        (*array).add_id(item);
-    }
+    assert!(!array.is_null());
+    (*array).add_id(item);
 }
 
 pub unsafe fn dc_array_add_ptr(array: *mut dc_array_t, item: *mut libc::c_void) {
     dc_array_add_uint(array, item as uintptr_t);
 }
 
 pub unsafe fn dc_array_get_cnt(array: *const dc_array_t) -> size_t {
-    if array.is_null() {
-        0
-    } else {
-        (*array).len()
-    }
+    assert!(!array.is_null());
+    (*array).len()
 }
 
 pub unsafe fn dc_array_get_uint(array: *const dc_array_t, index: size_t) -> uintptr_t {
-    if array.is_null() || index >= (*array).len() {
-        0
-    } else {
-        (*array).get_uint(index)
-    }
+    assert!(!array.is_null());
```

Comment:
Panicing on a null pointer changes the behaviour of the ffi, i think we should avoid this. If it's null, we should return a `0`

https://github.com/deltachat/deltachat-core/blob/e4ce281dc2c8cb1d0765f0e2e73f7846cbba3faa/src/dc_array.c#L252

Reply:
@r10s what do you think about this? Is it likely that we break android/ios with this change?

```
<Jikstra> we have to be careful with panics in the ffi just because array is a null pointer
<Jikstra> the api allowed us to do weird stuff, panicing will break all apps based on -core-rust
<link2xt[m]> yes, but such panics can be easily debugged
<link2xt[m]> if FFI call panics, it panics immediately
<link2xt[m]> and then you know you should not pass invalid value into it
<link2xt[m]> a simple if is enough to fix it in this case, but it may happen to be an actual bug
```

Reply:
See #291 for previous discussion of similar changes. No input from @r10s there, but it got merged.